#162528313: implement user signup

samorahno opened the pull request:
#### What does this PR do?
This PR creates a signup endpoint
#### Description of Task to be completed?
- The signup endpoint 
- JWT setup for generating token
- JsDoc setup and addition of style to some files
#### How should this be manually tested?
- Clone this repo  ```https://github.com/samorahno/i-reporter.git```
- Run ```npm run dev``` to start the server.
- Navigate to  ```localhost:9000/api/v1/auth/signup``` to create a new user
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
162528313
#### Screenshots
N/A
#### Questions
N/A

samorahno committed:
feat(signUp): implement user signup

 - signup a new user
 - setup eslint rule for jsdoc
 - implement JWT for generating token

 [Finishes #162528313]

iverenshaguy commented on server/controllers/userAuthController.js:
```diff
@@ -0,0 +1,88 @@
+/**
+ * 
+ */
+class userAuthControllerClass {   
```
Please remove the `Class`, it's obvious that it's a class. Just use `UserAuthController`. Also, classes should start with Uppercase (their file names too).
```suggestion
class UserAuthController {   
```

iverenshaguy commented on server/controllers/userAuthController.js:
```diff
@@ -0,0 +1,88 @@
+     * @returns {object} user object
+     */
+    static async createUser(req, res) {
+  
```
Please remove the new line

iverenshaguy commented on server/controllers/userAuthController.js:
```diff
@@ -0,0 +1,88 @@
+     */
+    static async createUser(req, res) {
+  
+      if (!req.body.fullname || !req.body.email || !req.body.password || !req.body.confirmPassword) {
```
Please make use of a validation middleware for validating your requests.

iverenshaguy commented on server/controllers/userAuthController.js:
```diff
@@ -0,0 +1,88 @@
+
+      try {
+        const { rows } = await dba.query(createQuery, values);
+        console.log(rows);
```
`console.log` statements should not be in the main code base. Please remove

iverenshaguy commented on server/controllers/userAuthController.js:
```diff
@@ -0,0 +1,88 @@
+        if (error.routine === '_bt_check_unique') {
+          return res.status(409).send({ status: 200, message: 'User with that EMAIL already exist' });
+        }
+        res.json({ status: 200, error });
```
Please `return` the response  e.g.
```suggestion
        return res.json({ status: 200, error });
```

iverenshaguy commented on server/db/tables.js:
```diff
@@ -81,10 +86,11 @@ const createInterventionTable = () => {
 
 /**
  * Drop Tables
+ * @returns {object} object
```
Since this doesn't return anything, you can just use void. eg
```suggestion
 * @returns {void}
```
This same thing applies to all functions that don't have the `return` keyword

iverenshaguy commented on server/controllers/userAuthController.js:
```diff
@@ -0,0 +1,88 @@
+/* eslint-disable indent */
+import moment from 'moment';
+import uuid from 'uuid';
+import 'babel-polyfill';
```
I can see that you used `babel-polyfill`. Can I see know why you are doing that? Also, since you have imported it in your index.js file, it would be available to other parts of the app. You do not need more than one import as done here

iverenshaguy commented on server/routes/userAuth.js:
```diff
@@ -0,0 +1,11 @@
+
+const router = express.Router();
+
+// http://localhost:9000/api/v1/auth/signup
```
???? Please what is this comment trying to achieve?

iverenshaguy commented on server/validation/userAuth.js:
```diff
@@ -0,0 +1,63 @@
+  hashPassword(password) {
+    return bcrypt.hashSync(password, bcrypt.genSaltSync(8));
+  },
+  /**
```
Please leave a new line before JSDocs
```suggestion

  /**
```

samorahno committed:
feat(signUp): implement user signup

 - signup a new user
 - setup eslint rule for jsdoc
 - implement JWT for generating token and LFA corrections on classes

 [Finishes #162528313]
